R/pairwise-comparisons.R

```diff
@@ -0,0 +1,594 @@
+
+  # remove "model" from 'by' before merging
+  by <- setdiff(by, "model")
+
+  overlap <- merge(a, b, by = by, allow.cartesian = TRUE)
+unique(overlap)
+
+  if (nrow(overlap) == 0) {
+    return(list(ratio = NA, pval = NA))
+  }
+
+  values_x <- overlap[[paste0(metric, ".x")]]
+  values_y <- overlap[[paste0(metric, ".y")]]
+
+  # calculate ratio to of average scores achieved by both models.
+  # thisshould be equivalent to theta_ij in Johannes document.
+  # ratio < 1 --> model1 is better.
+  # note we could also take mean(values_x) / mean(values_y), as it cancels out
+  ratio <- sum(values_x) / sum(values_y)
+
+  # test whether the ratio is significantly different from one
+  # equivalently, one can test whether the difference between the two values
+  # is significantly different from zero.
+  if (test_options$test_type[1] == "permutation") {
+    # adapted from the surveillance package
+    pval <- permutation_test(values_x, values_y,
+                             nPermutation = test_options$n_permutations,
+                             oneSided = test_options$oneSided,
+                             comparison_mode = "difference")
+  } else {
+    # this probably needs some more thought
+    # alternative: do a paired t-test on ranks?
+    pval <- wilcox.test(values_x, values_y, paired = TRUE)$p.value
+  }
+  return(list(mean_scores_ratio = ratio,
+              pval = pval))
+}
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+#' @title Plot Heatmap of Pairwise Comparisons
+#'
+#' @description
+#' Creates a heatmap of the ratios or pvalues from a pairwise comparison
+#' between models
+#'
+#' @param comparison_result A data.frame as produced by
+#' \code{\link{pairwise_comparison}}
+#' @param type character vector of length one that is either "mean_scores_ratio" or "pval".
+#' This denotes whether to visualise the ratio or the p-value of the
+#' pairwise comparison. Default is "mean_scores_ratio"
+#' @param smaller_is_good logical (default is TRUE) that indicates whether
+#' smaller or larger values are to be interpreted as 'good' (as you could just
+#' invert the mean scores ratio)
+#' @importFrom ggplot2 ggplot aes geom_tile geom_text labs coord_cartesian
+#' scale_fill_gradient2 theme_minimal element_text
+#' @importFrom data.table as.data.table
+#' @importFrom stats reorder
+#' @export
+#'
+#' @examples
+#' df <- data.frame(model = rep(c("model1", "model2", "model3"), each = 10),
+#'                  id = rep(1:10),
+#'                  interval_score = abs(rnorm(30, mean = rep(c(1, 1.3, 2), each = 10))),
+#'                  aem = (abs(rnorm(30))))
+#'
+#' res <- scoringutils::pairwise_comparison(df, baseline = "model1")
+#' plot_pairwise_comparison(res, smaller_is_good = TRUE)
+#'
+#' res <- pairwise_comparison(df)
+#' scoringutils::plot_pairwise_comparison(res, smaller_is_good = TRUE)
+#' scoringutils::plot_pairwise_comparison(res, smaller_is_good = TRUE, type = "pval")
+
+
+plot_pairwise_comparison <- function(comparison_result,
```

Comment:
it would be nice if you switched to S3 methods for plotting I think. We can chat about that if interested.